docs/DriverParseableOutput.rst

```diff
@@ -33,6 +33,29 @@ This allows the driver to pass as much information as it wants about the ongoing
 compilation, and programs which parse this data can decide what to use and what
 to ignore.
 
+
+Quasi-PIDs
+==========
+
+In previous versions of this format, certain fields labeled "pid" carry only
+legitimate process IDs (PIDs) from the underlying operating system. These PIDs
+are used by the driver to associate events that happen to jobs, such as signals
+when a job crashes with the file being compiled during that event. Real PIDs are
+always positive numbers, in keeping with POSIX semantics and the unsigned
+Process ID type on Windows.
+
+In more recent versions of this format, the Swift driver combines multiple jobs
+together into batches, running each batch in a single subprocess. In this mode,
+the driver writes messages describing each job as before, but assigns to each a
+so-called "quasi-PID" for the "pid" field, which does not correspond to an
+operating system process ID. Rather, a quasi-PID is a **negative number** that
+can be used **like** a PID, both as a unique identifier for each job that was
+run and in order to relate messages and tasks together, but it does not denote a
+real process. By using negative numbers for quasi-PIDs, we ensure that no two
+jobs will have the same PID: no batch mode job can have the same PID as a real
+process. Two jobs with the same PID would violate this uniqueness assumption.
```

Comment:
Thanks!

Reply:
Couldn't we use a separate field for this? Like a subtask UUID or something? I hate to mix semantically different data in the same field...